subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/tasks/JacocoReportBase.java

```diff
@@ -200,10 +205,13 @@ public void execute(Task task) {
     @Internal
     public FileCollection getAllClassDirs() {
```

Comment:
We'll need a bunch of test cases for the scenario:

- The extension defines no includes/excludes -> all classes directories are returned.
- The extension defines 1..n includes -> only the includes class directories are returned.
- The extension defines 1..n excludes -> only those class directories are returned that don't match the excludes.
- The extension defines 1..n overlapping includes and excludes -> returns the proper class directories.

We should these things on a unit test level but then also have integration tests that properly translate from extension to the added functionality in this method.